extensions/pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -77,12 +79,12 @@ func NewActuator(
 	atomicShootWebhookConfig *atomic.Value,
 	webhookServerNamespace string,
 	webhookServerPort int,
+	extendedAPIsForCleanup map[string]client.ObjectList,
```

Comment:
We discussed this with @plkokanov and agreed to keep this as is. The purpose of the generic actuators is to encapsulate logic that is common across all extensions so that they only have to pass their specific values (via arguments to this `NewActuator` function) instead of duplicating the code and reimplementing the same `n` times.

Obviously, this interface has grown quite big meanwhile and needs refactoring, but this is a different story.

Reply:
Maybe preferring @plkokanov approach is better, we can't assume that every extension force deletion logic will need to clean up the extended APIs, most of them can be deleted gracefully, so if in generic actuator we clean the extended APIs then all extensions that can be deleted gracefully has to overwrite `ForceDelete`, which is counter-productive. WDYT?